go.mod

```diff
@@ -17,7 +17,7 @@ require (
 	github.com/godbus/dbus v0.0.0-20181025153459-66d97aec3384 // indirect
 	github.com/gogo/protobuf v1.1.1
 	github.com/google/go-cmp v0.2.0 // indirect
-	github.com/mdlayher/vsock v0.0.0-20181130155850-676f733b747c
+	github.com/mdlayher/vsock v0.0.0-20180806202057-2b18bec03226
```

Comment:
Updating this dependency broke this PR, so I reverted it temporarily until we can figure out why. 

One time I got this in the agent logs: 
```
ime="2018-12-05T07:23:22Z" level=info msg="creating runc shim" id=
time="2018-12-05T07:23:22Z" level=info msg="listening to vsock" port=10789
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 779.41µs" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 153.551µs" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 2.145821ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 3.010051ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 15.113937ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 5.16732ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 16.969758ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 83.776148ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=info msg=create bundle=/run/containerd/io.containerd.runtime.v2.task/default/2151 id=2151
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 117.216µs" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 279.449µs" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=debug msg="Created listener"
time="2018-12-05T07:23:22Z" level=debug msg="Created listener"
time="2018-12-05T07:23:22Z" level=debug msg="Created listener"
time="2018-12-05T07:23:22Z" level=debug msg="Calling runc create"
time="2018-12-05T07:23:22Z" level=debug msg="Accepted connection"
time="2018-12-05T07:23:22Z" level=error msg="unable to accept vsock connection" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=debug msg="Accepted connection"
time="2018-12-05T07:23:22Z" level=error msg="unable to accept vsock connection" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=debug msg="Accepted connection"
time="2018-12-05T07:23:22Z" level=error msg="unable to accept vsock connection" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 398.084µs" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 7.669287ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 4.491574ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 22.458836ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 43.445515ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=debug msg="create succeeded" pid=206
time="2018-12-05T07:23:22Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:22Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=CREATED
time="2018-12-05T07:23:22Z" level=debug msg=start exec_id= id=2151
time="2018-12-05T07:23:22Z" level=debug msg=wait exec_id= id=2151
time="2018-12-05T07:23:22Z" level=debug msg="start succeeded" pid=206
time="2018-12-05T07:23:22Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:22Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=RUNNING
time="2018-12-05T07:23:22Z" level=debug msg=resize_pty exec_id= id=2151
time="2018-12-05T07:23:22Z" level=debug msg="resize_pty succeeded"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 62.182873ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 80.644968ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 478.584091ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:22Z" level=error msg="ttrpc: failed accept; backoff 539.11079ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:23Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:23Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=RUNNING
time="2018-12-05T07:23:23Z" level=error msg="ttrpc: failed accept; backoff 853.353331ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:24Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:24Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=RUNNING
time="2018-12-05T07:23:24Z" level=error msg="ttrpc: failed accept; backoff 324.778273ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:24Z" level=error msg="ttrpc: failed accept; backoff 138.149956ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:24Z" level=error msg="ttrpc: failed accept; backoff 693.774911ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:25Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:25Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=RUNNING
time="2018-12-05T07:23:25Z" level=error msg="ttrpc: failed accept; backoff 183.515637ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:25Z" level=error msg="ttrpc: failed accept; backoff 821.866378ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:26Z" level=debug msg=state exec_id= id=2151
time="2018-12-05T07:23:26Z" level=debug msg="state succeeded" bundle=/container id=2151 pid=206 status=RUNNING
time="2018-12-05T07:23:26Z" level=error msg="ttrpc: failed accept; backoff 342.231109ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:26Z" level=error msg="ttrpc: failed accept; backoff 797.652072ms" error="resource temporarily unavailable"
time="2018-12-05T07:23:27Z" level=debug msg=state exec_id= id=2151
...
```
but all other times it fails at startup with:
`
ctr: ttrpc: client shutting down: read vsock:host(2):1415: transport endpoint is not connected: unknown
`

The error seems to be coming from ttrpc [here](https://github.com/containerd/ttrpc/blob/2a805f71863501300ae1976d29f0454ae003e85a/server.go#L106) cause by errors from calling Accep (resource temporarily unavailable). I think the only change between the two revisions is this PR https://github.com/mdlayher/vsock/pull/20 . I'm not sure exactly how that change causes these errors. It'd be interesting if someone else could verify the behavior in case it's something local to my box. 


The steps to test the updated dependency:
1. modify go.mod to add `github.com/mdlayher/vsock v0.0.0-20181130155850-676f733b747c` for the vsock version instead of `github.com/mdlayher/vsock v0.0.0-20180806202057-2b18bec03226`
1. `go mod tidy`
1. make clean all
1. copy over runtime to path and agent to the image
1. ctr run an image like : `% sudo ./ctr run --runtime aws.firecracker --snapshotter firecracker-snapshotter --tty docker.io/library/alpine:latest 2165 /bin/sh`. I tried both with/without --tty and it didn't seem to make a difference.

Reply:
I'll take a closer look on this, feel free to revert this commit if it blocks your PR.

Reply:
I'd rather figure out the core issue than revert if possible.

Reply:
I can confirm this problem when building with `github.com/mdlayher/vsock v0.0.0-20180806202057-2b18bec03226`

The problem is not reproducible with the previous version.

Reply:
@nmeyerhans can you clarify? you see error messages or it just doesn't work?
This messages are caused by non-blocking behavior I recently [added](https://github.com/mdlayher/vsock/commit/fda437ecbf04836c30bfb8c397ca7383c5d01a7e) to `vsock` package, so `Accept` no longer blocks, but just returns temporary error https://github.com/containerd/ttrpc/blob/master/server.go#L95
Except these annoying error messages, `agent` should work as expected.